src/SolverAlsWrmf.cpp

```diff
@@ -304,11 +304,21 @@ double als_explicit_float(const Rcpp::S4 &m_csc_r,
 }
 
 template <class T>
-void initialize_biases(const dMappedCSC& ConfCSC,
-                       const dMappedCSC& ConfCSR,
-                       arma::Col<T>& user_bias,
-                       arma::Col<T>& item_bias,
-                       T lambda, bool non_negative) {
+double initialize_biases(const dMappedCSC& ConfCSC,
+                         const dMappedCSC& ConfCSR,
+                         arma::Col<T>& user_bias,
+                         arma::Col<T>& item_bias,
+                         T lambda, bool non_negative) {
+  /* Robust mean calculation */
+  double glob_mean = 0;
+  for (size_t ix = 0; ix < ConfCSC.nnz; ix++)
+    glob_mean += (ConfCSC.values[ix] - glob_mean) / (double)(ix+1);
+#pragma omp simd
```

Comment:
we need to add conditional `#ifdef _OPENMP` as it won't compile on certain platforms